test/e2e/basic_ops_test.go

```diff
@@ -47,18 +48,7 @@ var _ = Describe("basic", func() {
 
 	It("should control power cycle of BMH though annotations", func() {
 		By("creating a secret with BMH credentials")
-		bmcCredentials := corev1.Secret{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      "bmc-credentials",
-				Namespace: namespace.Name,
-			},
-			StringData: map[string]string{
-				"username": bmcUser,
-				"password": bmcPassword,
-			},
-		}
-		err := clusterProxy.GetClient().Create(ctx, &bmcCredentials)
-		Expect(err).NotTo(HaveOccurred())
+		CreateBMHCredentialsSecret(ctx, clusterProxy.GetClient(), namespace.Name, secretName, bmcUser, bmcPassword)
 
 		By("creating a BMH")
 		bmh := metal3api.BareMetalHost{
```

Comment:
Do you mean "creating a BMH" into a function?

Reply:
I'm not convinced this would be a good idea. We would basically need parameters for everything since we are testing all configurations for the BareMetalHosts. Then it is easier to just create it directly

Reply:
@maxrantil  Yeah that what I was thinking so far seems we are mostly just changing annotations if we can pass them as parameters would reduce a bit the duplication but probably in other cases this becomes parameters for everything as @lentzi90 stated.